common/txmgr/confirmer.go

```diff
@@ -294,178 +260,144 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Pro
-// the gas bumping cycle. This is necessary in rare cases (e.g. Polygon) where
-// network conditions are extremely hostile.
-//
-// For example, assume the following scenario:
-//
-// 0. We are connected to multiple primary nodes via load balancer
-// 1. We send a transaction, it is confirmed and, we get a receipt
-// 2. A new head comes in from RPC node 1 indicating that this transaction was re-org'd, so we put it back into unconfirmed state
-// 3. We re-send that transaction to a RPC node 2 **which hasn't caught up to this re-org yet**
-// 4. RPC node 2 still has an old view of the chain, so it returns us "sequence too low" indicating "no problem this transaction is already mined"
-// 5. Now the transaction is marked "confirmed_missing_receipt" but the latest chain does not actually include it
-// 6. Now we are reliant on the Resender to propagate it, and this transaction will not be gas bumped, so in the event of gas spikes it could languish or even be evicted from the mempool and hold up the queue
-// 7. Even if/when RPC node 2 catches up, the transaction is still stuck in state "confirmed_missing_receipt"
-//
-// This scenario might sound unlikely but has been observed to happen multiple times in the wild on Polygon.
-func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CheckConfirmedMissingReceipt(ctx context.Context) (err error) {
-	attempts, err := ec.txStore.FindTxAttemptsConfirmedMissingReceipt(ctx, ec.chainID)
-	if err != nil {
-		return err
-	}
-	if len(attempts) == 0 {
-		return nil
-	}
-	ec.lggr.Infow(fmt.Sprintf("Found %d transactions confirmed_missing_receipt. The RPC node did not give us a receipt for these transactions even though it should have been mined. This could be due to using the wallet with an external account, or if the primary node is not synced or not propagating transactions properly", len(attempts)), "attempts", attempts)
-	txCodes, txErrs, broadcastTime, txIDs, err := ec.client.BatchSendTransactions(ctx, attempts, int(ec.chainConfig.RPCDefaultBatchSize()), ec.lggr)
-	// update broadcast times before checking additional errors
-	if len(txIDs) > 0 {
-		if updateErr := ec.txStore.UpdateBroadcastAts(ctx, broadcastTime, txIDs); updateErr != nil {
-			err = fmt.Errorf("%w: failed to update broadcast time: %w", err, updateErr)
+// CheckForConfirmation fetches the mined transaction count for each enabled address and marks transactions with a lower sequence as confirmed and ones with equal or higher sequence as unconfirmed
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CheckForConfirmation(ctx context.Context, head types.Head[BLOCK_HASH]) error {
+	for _, fromAddress := range ec.enabledAddresses {
+		minedTxCount, err := ec.client.SequenceAt(ctx, fromAddress, nil)
+		if err != nil {
+			return fmt.Errorf("unable to fetch mined transaction count for address %s: %w", fromAddress.String(), err)
```

Comment:
It makes more sense to use `continue` and create a `multierr` here but since multiple enabled addresses are rarely used and there is an upcoming refactoring, we can skip for now.

Reply:
I think this was small enough of a change that was worth to address now. Even though it's rare, don't want to risk introducing any issues.

Reply:
The change looks good, but I would say the complete solution would also include handling the returned error because now the `processHead` loop will exit even if a single address fails. One solution could be to return the successful addresses and pass them to `ProcessStuckTransactions` and `RebroadcastWhereNecessary`. Alternatively, we can argue that we want to run `ProcessStuckTransactions` and `RebroadcastWhereNecessary` regardless of the error returned from `CheckForConfirmation`. I think for now this looks ok, so let's not waste too much time on it.